service/controller/resource/ensurecpcrs/create.go

```diff
@@ -94,13 +97,6 @@ func (r *Resource) createAWSControlPlaneCR(ctx context.Context, cr infrastructur
 				label.Release:         key.ReleaseVersion(&cr),
 			},
 			Name: id,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					APIVersion: infrastructurev1alpha2.NewG8sControlPlaneTypeMeta().APIVersion,
-					Kind:       infrastructurev1alpha2.NewG8sControlPlaneTypeMeta().Kind,
-					Name:       id,
-				},
-			},
```

Comment:
FYI I dropped the owner reference as this did not work well. You apparently must provide the UID of the Kubernetes runtime object, which would be the `G8sControlPlane`, which does not exist at this point. So you need to update one of the existing objects in any case when you want to ensure the link towards both sides. I do not see me doing this now. The owner reference was a nice to have and now that it already causes trouble I do not see any reason to further pursue it. In case we want that we do it later. Note that Node Pools do also not link back to the `MachineDeployment` CR. So whatever.

Reply:
Understood. I think we can do without it, if it was troublesome.